lib/src/main/kotlin/com/swmansion/starknet/data/types/Execution.kt

```diff
@@ -97,7 +97,23 @@ data class ExecutionParams(
 )
 
 @JvmSynthetic
-internal fun callsToExecuteCalldata(calls: List<Call>): List<Felt> {
+private fun callsToExecuteCalldataCairo1(calls: List<Call>): List<Felt> {
+    val result = mutableListOf<Felt>()
+
+    result.add(Felt(calls.size))
+
+    for (call in calls) {
+        result.add(call.contractAddress)
+        result.add(call.entrypoint)
+        result.add(Felt(call.calldata.size))
```

Comment:
are you sure size should be included here?

Reply:
From what I understand about execute calldata in these new contracts, that's how it should be constructed now. Without calldata size, there wouldn't be an option to determine how many calldata arguments are in the call

Reply:
I can confirm that's the correct implementation 👍